core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java

```diff
@@ -703,6 +708,60 @@ public static void toArmeria(io.netty.handler.codec.http.HttpHeaders inHeaders,
         }
     }
 
+    private static String maybeTransformAbsoluteUri(
+            String path, Function<? super String, String> absoluteUriTransformer) throws URISyntaxException {
+
+        if (isValidHttp2Path(path)) {
+            return path;
+        }
+
+        if (!isAbsoluteUri(path)) {
+            throw newInvalidPathException(path);
+        }
+
+        final String newPath;
+        try {
+            newPath = absoluteUriTransformer.apply(path);
+        } catch (Exception e) {
+            warnExceptionThrowingAbsoluteUriTransformer(e);
+            throw newInvalidPathException(path);
+        }
+
+        if (newPath == null) {
+            warnNullReturningAbsoluteUriTransformer();
+            throw newInvalidPathException(path);
+        }
+
+        if (path.equals(newPath) || !isValidHttp2Path(newPath)) {
```

Comment:
minor question) Is it possible to just check `isValidHttp2Path`? 
I'm thinking about the worst case where `String` isn't stored in the constant pool and each character has to be compared. I'm wondering if just doing the `isValidHttp2Path` check is faster/simpler.

```suggestion
        if (!isValidHttp2Path(newPath)) {
```

Reply:
Agreed. Let me push your suggestion.